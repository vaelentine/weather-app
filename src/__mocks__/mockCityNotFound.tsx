export const mockCityNotFound =
{
  cod: '404',
  message: 'city not found'
}
