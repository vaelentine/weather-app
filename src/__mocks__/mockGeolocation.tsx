export const mockGeolocation = {
    getCurrentPosition: jest.fn()
        .mockImplementationOnce(success => Promise.resolve(success({
            coords: {
                latitude: 46.378333,
                longitude: 13.836667,
            },
        }))),
    watchPosition: jest.fn(),
}

// global.navigator.geolocation = mockGeolocation