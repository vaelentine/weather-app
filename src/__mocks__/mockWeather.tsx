const mockWeather =
  {
    coord: {
      lon: -122.6762,
      lat: 45.5234
    },
    weather: [
      {
        id: 800,
        main: 'Clear',
        description: 'clear sky',
        icon: '01n'
      }
    ],
    base: 'stations',
    main: {
      temp: 291.01,
      feels_like: 290.26,
      temp_min: 286.82,
      temp_max: 293.11,
      pressure: 1013,
      humidity: 54
    },
    visibility: 10000,
    wind: {
      speed: 0.89,
      deg: 313,
      gust: 4.02
    },
    clouds: {
      all: 1
    },
    dt: 1630560308,
    sys: {
      type: 2,
      id: 2008548,
      country: 'US',
      sunrise: 1630503161,
      sunset: 1630550935
    },
    timezone: -25200,
    id: 5746545,
    name: 'Portland',
    cod: 200
  }

export default mockWeather
