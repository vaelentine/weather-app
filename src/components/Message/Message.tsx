import React from "react";

interface MessageProps {
    messageData:any
}

const Message: React.FC<MessageProps> = ({ messageData }) => {
  return (
    <div className="errorMessage">
        API Error: {messageData.message}
    </div>
  )
}

export default Message
