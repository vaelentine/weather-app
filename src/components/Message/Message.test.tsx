/* eslint-disable no-undef */
import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Message from './Message'
import { mockCityNotFound } from '__mocks__/mockCityNotFound'

describe('Search Component', () => {
  it('renders correctly', () => {
    const data = mockCityNotFound
    const component = render(<Message messageData={ data }/>)
    expect(component.asFragment()).toMatchSnapshot()
  })
  it('displays the message correctly', () => {
    const data = mockCityNotFound
    const component = render(<Message messageData={ data }/>)
    const errorMessage = component.getByText(/API Error/)
    const errorText = new RegExp(data.message)
    expect(errorMessage).toHaveTextContent(errorText)
  })
})
