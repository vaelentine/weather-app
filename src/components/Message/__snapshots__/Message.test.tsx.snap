// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="errorMessage"
  >
    API Error: city not found
  </div>
</DocumentFragment>
`;
